packages/core/src/lib/select/searchable-select.svelte

```diff
@@ -0,0 +1,273 @@
+};
+
+const handleSelect = (option: string) => {
+  open = false;
+
+  if (value === option) {
+    return;
+  }
+
+  value = option;
+
+  dispatch('change', value);
+  dispatch('input', value);
+};
+
+const handleFocus = () => {
+  if (open || disabled) {
+    return;
+  }
+
+  open = true;
+};
+
+const handleOptionFocus = (index: number) => {
+  if (!keyboardControlling) {
+    navigationIndex = index;
+  }
+};
+
+const handleClickOutside = (event: Event) => {
+  if (clickedOutside(event, wrapper)) {
+    closeMenu();
+  }
+};
+
+const handleMouseLeave = () => (navigationIndex = -1);
+const handleButtonClick = () => dispatch('buttonclick');
+
+$: {
+  if (doesSearch && options.length > 0 && searchTerm) {
+    dispatch('search', searchTerm);
+    sortedOptions = matchAndSortEntries(options, searchTerm, isReduceSort);
+  } else {
+    sortedOptions = options;
+  }
+
+  searchedOptions = applySearchHighlight(
+    sortedOptions,
+    doesSearch ? searchTerm : ''
+  );
+}
+
+$: {
+  if (!open && exact && value && !options.includes(value)) {
+    value = undefined;
+    dispatch('change', value);
+    dispatch('input', value);
+  }
+}
+</script>
+
+<svelte:body on:click={handleClickOutside} />
+<div
+  bind:this={wrapper}
+  class="relative flex w-full"
+>
+  <div class="flex w-full">
+    <input
+      bind:this={input}
+      bind:value
+      readonly={disabled ? true : undefined}
+      aria-disabled={disabled ? true : undefined}
+      type="text"
+      class={cx(
+        'h-[30px] w-full grow appearance-none border py-1.5 pl-2 pr-1 text-xs leading-tight outline-none',
+        {
+          'border-light hover:border-gray-6 focus:border-gray-9 bg-white':
+            !disabled && !isError && !isWarn,
+          'border-disabled-light focus:border-disabled-dark bg-disabled-light text-disabled-dark cursor-not-allowed':
+            disabled,
+          'border-warning-bright hover:outline-warning-bright focus:outline-warning-bright hover:outline-[1.5px] hover:-outline-offset-1 focus:outline-[1.5px] focus:-outline-offset-1':
+            isWarn,
+          'border-danger-dark hover:outline-danger-dark focus:outline-danger-dark hover:outline-[1.5px hover:-outline-offset-1 focus:outline-[1.5px] focus:-outline-offset-1':
+            isError,
+        }
+      )}
+      {...$$restProps}
+      on:input|preventDefault={handleInput}
+      on:keydown={handleKeyDown}
+      on:focus={handleFocus}
+      on:mousemove={() => setKeyboardControl(false)}
+    />
+
+    <Icon
+      name="chevron-down"
+      cx={[
+        'text-gray-6 absolute top-1.5 right-2 transition',
+        { 'rotate-180': open },
+      ]}
+    />
+  </div>
+  <SelectMenu
+    bind:element={menu}
+    bind:heading
+    bind:button
+    on:buttonclick={handleButtonClick}
+    on:mouseleave={handleMouseLeave}
+    {open}
+  >
+    {#if sortedOptions.length > 0}
+      {#each searchedOptions as { search, option }, index (option)}
+        <li role="presentation">
+          <button
```

Comment:
The button is focusable by tab natively with this layout, so it shouldn't require a `tabindex`. If we wanted to skip the `button` and just use the `li` as a `menuitem` we would need a `tabindex` there and would need rely on `click` and `keydown` handlers instead of just the `button`'s click.

Reply:
Per [MDN's docs on `menu`](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/menu_role):

> [tabindex](https://developer.mozilla.org/en-US/docs/Web/HTML/Global_attributes#tabindex) attribute
The `menu` container has `tabindex` set to `-1` or `0` and each item in the menu has tabindex set to `-1`.

Inside a vertical `menu`, it seems like the recommended keyboard interaction (of MDN) is to use up and down arrows to navigate menu options, not tab

Reply:
You can navigate this with the up and down arrow keys and you cannot use tab. The menu is a `ul` which is non-interactive which is equal to `tabindex` of `-1` so I can add it but shouldn't change the behavior. You can use to tab to exit it along with escape, but because we capture and handle the tab event it is impossible to navigate to the buttons with it. I can set a `tabindex` of `-1` on the buttons but similarly it won't change the behavior.

Reply:
Interesting, I just tried removing the buttons as well and moving everything to the `li` and it all behaves as expected but setting the `height` on the `li` doesn't work:

<img width="362" alt="Screenshot 2023-08-25 at 3 55 56 PM" src="https://github.com/viamrobotics/prime/assets/1930371/cdfcb6df-4664-4f12-b6b1-923a28108945">